### ISSUE TITLE
Error message

### ISSUE THREAD

**Issue by vera-rykalina** [DMU-lilab/pTrimmer#27]
Hi,

I am just testing the tool on my FASTQ files and always getting an error "buffer overflow detected". What does it mean?

Do FASTQ files need any preprocessing (e.g. a header change)?  

Should the tool also work when there are only four overlapping regions with long insertion sizes, e.g. like here https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3502977/ 

Thanks in advance.

**Comment by XLZH** [comment]
@vera-rykalina could you please provide a small subset of reads to reproduce the error? For the example you mentioned above, I never test it before, but I think it can work well if the primer-pair is given

**Comment by vera-rykalina** [comment]
Thanks for your quick reply.  I will do it on Monday when I am the institute.
By the way, I installed the tool on HPC via conda which is supposed to give me version 1.4.0. The installed version is 1.3.3 although according to the help message.

**Comment by XLZH** [comment]
The conda package is not maintained by our group, therefore, it is not updated now. I advise you download the v1.4.0 here and install it according to the INSTALL, because I have refactored the part of handling FASTQ reads.

**Comment by vera-rykalina** [comment]
I believe, it was an outdated version of the SW that caused the issue. I could get all the output files, using version 1.4.0. However, all reads in the output files contain only N's. 
 
@M02885:307:000000000-CTYL9:1:1101:14863:1458 1:N:0:TAAGGCTA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:22916:1474 1:N:0:TAAGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:19462:1492 1:N:0:TATGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:22624:1502 1:N:0:TATGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:18965:1510 1:N:0:TAAGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!

I think the tool cannot be applied for the sequencing design I mentioned above, as the primer sequences have a varied location within the reads (see the screenshot): 
![primers_read1](https://github.com/user-attachments/assets/f3054a8d-221b-42d5-bcf9-594d7b840728)

**Comment by XLZH** [comment]
You are right, pTrimmer may not be a proper tool for your condition, because the forward primer sequence usually exists in the begging of the read in target/amplicon sequencing data!在 2024年7月15日，14:51，vera-rykalina ***@***.***> 写道：﻿
I believe, it had been an outdated version that caused the issue. I could get all the output files, using version 1.4.0. However, all reads in the output files contain only N's.
@M02885:307:000000000-CTYL9:1:1101:14863:1458 1:N:0:TAAGGCTA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:22916:1474 1:N:0:TAAGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:19462:1492 1:N:0:TATGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:22624:1502 1:N:0:TATGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
@M02885:307:000000000-CTYL9:1:1101:18965:1510 1:N:0:TAAGGCGA+CTATTAAT
NNNNNNNNNNNNNNNNNNNN
+
!!!!!!!!!!!!!!!!!!!!
I think the tool cannot be applied for the sequencing design I mentioned above, as the primer sequences have a varied location within the reads (see the screenshot):
primers_read1.PNG (view on web)
